src/components/App.tsx

```diff
@@ -309,8 +326,17 @@ class App extends React.Component<any, AppState> {
       } else {
         scene = await loadScene(null);
       }
+
       if (scene) {
         this.syncActionResult(scene);
+        if (scene.appState) {
+          const { showZenMode } = scene.appState;
+          if (showZenMode) {
+            this.setState({
+              showZenModeAlertDialog: true,
```

Comment:
I'm not really sure how this is enabled but I haven't been able to see this dialog. I typed alt-z in order to enable zen mode.

Zen mode suggests that it relaxes you. Showing a big dialog is going against that.

My understanding is that the rationale behind the dialog is that if you hit the keybinding by surprise, you won't be able to go back. Instead of showing a big dialog, we should put a UI affordance to exit the zen mode. Right now looks like you keep the keybinding icon but that's not super clear why, the dialog has tons of keybindings. I would put a small X somewhere in the ui to bring back the rest of the UI so you can avoid this dialog altogether.

Reply:
This dialog only appears when you are already using zen mode and its set in local storage and you refresh the page so this dialog will appear to inform the user whether they want to exit or enter the zen mode.
Typing Alt+z will take you to zen mode but since we set in local storage so this will help the user to come out of it if they want to. 
we had the option of adding a small icon somewhere in the UI as an option but since its distraction-free mode so didn't want to add any UI element as just with short cut keys it looks smoother.

Reply:
@ad1992 What I had in mind for the "dialog" was more like what we show when you scroll away from content: 

![image](https://user-images.githubusercontent.com/5153846/79686012-5d690100-823d-11ea-9e54-96ed31542da4.png)

And it would go away after ~3 seconds after load.

My bad I didn't explain this, and my brain completely missed that you suggested an actual dialog with a call for action, which is not what we want (we also don't have any public document explaining our design philosophy yet).

Replacing the keyboard-shortcuts icon with a small `exit zen mode` or `x` button as @vjeux suggested seems fine.

Reply:
Ohh, I get it now. So I have one confusion now
I can replace the keyboard shortcuts. with "exit zen mode " as I m not sure if `x` will be super intuitive so explicitly mentioning "exit zen mode" looks better. Now since we have exit zen mode button right in front always then the logic for checking on page refresh is not needed and the fading away of a button is also not needed. Did I get it right @dwelle?

Reply:
Yes, if you go with the `exit` button then we don't need the logic for refresh. :+1: